language/lexer/lexer.go

```diff
@@ -106,24 +102,29 @@ func Lex(s *source.Source) Lexer {
 
 // Reads an alphanumeric + underscore name from the source.
 // [_A-Za-z][_0-9A-Za-z]*
-func readName(source *source.Source, position int) Token {
+// position: Points to the byte position in the byte array
+// runePosition: Points to the rune position in the byte array
+func readName(source *source.Source, position, runePosition int) Token {
 	body := source.Body
 	bodyLength := len(body)
-	end := position + 1
+	endByte := position + 1
+	endRune := runePosition + 1
 	for {
-		code, n := runeAt(body, end)
-		if (end != bodyLength) &&
+		code, n := runeAt(body, endByte)
+		if (endByte != bodyLength) &&
+			n == 1 && // only ASCII chars are allowed to be names
```

Comment:
The n == 1 check is superfluos here, since the next and condition only checks for char codes which are all single-byte characters.